engine/runtime/src/main/java/org/enso/interpreter/runtime/data/EnsoDateTime.java

```diff
@@ -138,6 +146,27 @@ public EnsoTimeZone zone() {
     return new EnsoTimeZone(dateTime.getZone());
   }
 
+  @Builtin.Method(name = "plus_builtin", description = "Adds a duration to this date time")
+  @Builtin.Specialize
+  public EnsoDateTime plus(Object durationObject, InteropLibrary interop) {
+    return new EnsoDateTime(dateTime.plus(ensureDuration(durationObject, interop)));
+  }
+
+  @Builtin.Method(name = "minus_builtin", description = "Subtracts a duration from this date time")
+  @Builtin.Specialize
```

Comment:
Changing builtins will very likely need changes to reflective files like in #3755 - please run `sbt engine-runner-native/buildNativeImage` to verify it builds properly. CCing @hubertp , @mwu-tow .

Reply:
To maybe complement @JaroslavTulach's comment - once it finishes (it will take a while) you should be able to execute `./runner --run engine/runner-native/src/test/resources/Factorial.enso 6` without errors